form-builder/index.html

```diff
@@ -0,0 +1,956 @@
+                newQuestion.exclusiveMinimum(question.exclusiveMinimum());
+                newQuestion.exclusiveMaximum(question.exclusiveMaximum());
+                newQuestion.multiChoice(question.multiChoice());
+                newQuestion.multiLineText(question.multiLineText());
+                newQuestion.required(question.required());
+                newQuestion.forAnnotation(question.forAnnotation());
+                this.questions.push(newQuestion);
+            }
+
+        }
+
+        class FormViewModel {
+            constructor() {
+                this.formId = ko.observable('');
+                this.title = ko.observable('');
+                this.description = ko.observable('');
+                this.associatedEntity = ko.observable('');
+                this.sections = ko.observableArray([]);
+                this.forAnnotationOptions = ko.observableArray([]);
+
+                ko.computed(() => {
+                    this.forAnnotationOptions(this.sections().reduce((acc, section) => {
+                        section.questions().forEach(question => {
+                            if (question.type() === 'initials' || question.type() === 'signature') {
+                                acc.push(question.questionId());
+                            }
+                        });
+                        return acc;
+                    }, []));
+                });
+
+            }
+
+            addSection() {
+                const sectionId = this.sections().length + 1;
+                this.sections.push(new SectionViewModel(sectionId));
+                this.renumberSections();
+            }
+
+            removeSection(section) {
+                if (confirm('Are you sure you want to remove this section?')) {
+                    this.sections.remove(section);
+                } this.renumberSections();
+            }
+
+            duplicateSection(section) {
+                //duplicate the section and add it to the form
+                const newSection = new SectionViewModel(this.sections().length + 1);
+                newSection.title(section.title());
+                newSection.subtitle(section.subtitle());
+                newSection.repeatable(section.repeatable());
+                newSection.newPage(section.newPage());
+                newSection.sectionHelpText(section.sectionHelpText());
+                section.questions().forEach(question => {
+                    const newQuestion = new QuestionViewModel(newSection.sectionId(), question.questionId());
+                    newQuestion.title(question.title());
+                    newQuestion.subtitle(question.subtitle());
+                    newQuestion.questionHelpText(question.questionHelpText());
+                    newQuestion.type(question.type());
+                    newQuestion.choices(question.choices().map(choice => ko.observable(choice())));
+                    newQuestion.minimum(question.minimum());
```

Comment:
The `Min/Max` or `Exclusive Min / Exclusive Max` should be the actual values instead of the string value.